src/Haskell/Ide/Engine/Plugin/GhcMod.hs

```diff
@@ -290,6 +293,66 @@ cmp a b
 isSubRangeOf :: Range -> Range -> Bool
 isSubRangeOf (Range sa ea) (Range sb eb) = sb <= sa && eb >= ea
 
+
+
+data DocumentPosition =
```

Comment:
Shouldn't this be an instance of `TextDocumentPositionParams`

```typescript
interface TextDocumentPositionParams {
	/**
	 * The text document.
	 */
	textDocument: TextDocumentIdentifier;

	/**
	 * The position inside the text document.
	 */
	position: Position;
}
```

as per https://microsoft.github.io/language-server-protocol/specification

This is made available via `haskell-lsp-types`, see https://github.com/alanz/haskell-lsp/blob/406b9e0e795b5e9c2ae4a2ab29b2339c29fa0354/haskell-lsp-types/src/Language/Haskell/LSP/TH/DataTypesJSON.hs#L1008